include/boost/multiprecision/cpp_int/misc.hpp

```diff
@@ -20,17 +20,19 @@
 #include <boost/multiprecision/detail/bitscan.hpp> // lsb etc
 #include <boost/multiprecision/detail/hash.hpp>
 #include <boost/multiprecision/detail/no_exceptions_support.hpp>
-#include <numeric> // std::gcd
+
+#ifdef BOOST_MP_MATH_AVAILABLE
+#include <boost/math/special_functions/next.hpp>
+#endif
+
 #include <type_traits>
 #include <stdexcept>
 #include <cmath>
 
-#ifndef BOOST_MP_STANDALONE
-#include <boost/integer/common_factor_rt.hpp>
-#endif
+#if (defined(__cpp_lib_gcd_lcm) && (__cpp_lib_gcd_lcm >= 201606L)) && (!defined(BOOST_HAS_INT128) || !defined(__STRICT_ANSI__))
```

Comment:
I wonder if we really need the `&& (!defined(BOOST_HAS_INT128) || !defined(__STRICT_ANSI__))` part?  Ah wait... no std::gcd and std::lcm for __int128?  I know we have our own gcd/lcm as a fall-back of last resort, but they're not actually very performant (I suspect anyway) compared to the full Boost versions.  Or perhaps I'm wrong on that?

In any case we need to get these CI tests fixed up before we can do much of anything... :(